x-pack/plugins/ml/public/application/datavisualizer/index_based/components/search_panel/search_panel.tsx

```diff
@@ -74,20 +90,30 @@ export const SearchPanel: FC<Props> = ({
 
   const searchHandler = (query: Query) => {
     let filterQuery;
-    if (query.language === SEARCH_QUERY_LANGUAGE.KUERY) {
-      filterQuery = esKuery.toElasticsearchQuery(
-        esKuery.fromKueryExpression(query.query),
-        indexPattern
-      );
-    } else if (query.language === SEARCH_QUERY_LANGUAGE.LUCENE) {
-      filterQuery = esQuery.luceneStringToDsl(query.query);
-    } else {
-      filterQuery = {};
-    }
+    try {
+      if (query.language === SEARCH_QUERY_LANGUAGE.KUERY) {
+        filterQuery = esKuery.toElasticsearchQuery(
+          esKuery.fromKueryExpression(query.query),
+          indexPattern
+        );
+      } else if (query.language === SEARCH_QUERY_LANGUAGE.LUCENE) {
+        filterQuery = esQuery.luceneStringToDsl(query.query);
+      } else {
+        filterQuery = {};
+      }
 
-    setSearchQuery(filterQuery);
-    setSearchString(query.query);
-    setSearchQueryLanguage(query.language);
+      setSearchQuery(filterQuery);
+      setSearchString(query.query);
+      setSearchQueryLanguage(query.language);
+    } catch (e) {
+      console.log('Invalid syntax', e); // eslint-disable-line no-console
+      const toastNotifications = getToastNotifications();
+      const notification =
+        query.language === SEARCH_QUERY_LANGUAGE.KUERY
+          ? kqlSyntaxErrorMessage
+          : luceneSyntaxErrorMessage;
+      toastNotifications.addDanger(notification);
```

Comment:
This is fine for now, but for a follow-up would be nice to get access to the error text in the toast, in a similar style to Discover:

![image](https://user-images.githubusercontent.com/7405507/77090124-a0e20c80-69fe-11ea-9d3b-2908ebc5b067.png)